src/usecases/updateCounter.ts

```diff
@@ -0,0 +1,41 @@
+import { Ports } from "~/adapter";
+import { RegisteredCourse, RegisteredCourseWithoutID } from "~/api/@types";
+
+/**
+ * 出欠のカウントを更新する。
+ *
+ * @param {string} id - course_id
+ * @param {RegisteredCourseWithoutID} usedCourse - 講義データ
+ * @param {string} key - attendance, absence or late
+ * @param {number} value - key's count
+ * @return {number}- key's count
+ */
+
+export const updateCounter = (
+  id: string,
+  usedCourse: RegisteredCourseWithoutID,
+  key: string,
+  value: number
+) => async ({ api, store, storage }: Ports) => {
+  console.log("updateCounter");
+  try {
+    /** API */
+    // usedCourse[key] = value;
+    // const { status, body: currentCourse } = await api.registered_courses._id(id).put({ body: usedCou });
+    // if (status != 200) throw new Error("講義の情報を更新できませんでした。");
+    // store.commit("updateCourse", currentCourse);
+    // storage.set("courses");
+    // return currentCourse[key];
+
+    /** APIの代わり */
+    const currentCourse = {} as RegisteredCourse;
+    Object.assign(currentCourse, store.getters.course(id));
+    currentCourse[key] = value;
+    store.commit("updateCourse", currentCourse);
+    storage.set("courses");
+    return currentCourse[key];
+  } catch (error) {
+    console.log(error);
+    throw new Error("講義の情報を更新できませんでした。");
```

Comment:
throwされたものを折角catchしたのにまたthrowしてる
console.errorぐらいでいいんじゃない？
```suggestion
    console.error(error);
```